### DISCUSSION TITLE
Get relations of an object

### DISCUSSION THREAD

**Discussion by gajanthini** [Volue-Public/energy-mesh-python#302] [Q&A] [answered]
Hi, 
I have a question, and I think the best way to explain this is by using a specific example. I would like to receive relations connected to an object, say Aurland1 and the connections it has to reservoirs Viddalsvatn and Vassbygdvatn as shown in hydro topology:
![hydro_topology](https://user-images.githubusercontent.com/111579046/189143008-b2c65996-e581-4273-ad9b-a531919dd611.PNG)

In general, I would like to find a way of outputting all sorts of connections, say waterways, floodways, hydroplants, generators, etc.. in such way that I can traverse a model without being dependent on being in Nimbus/Mesh looking at UUID or paths defined there. 

When using the "get_object function" on Aurland1, it outputs a list of relevant attributes, however, I have not been able to understand which of these attributes that actually gives me the information about the relations to the two hydroplants. I would have assumed that the "to_HydroPlants" attribute would, but this attribute only gives me the information of name, id and path of the attribute itself. How can I read information regarding relations to an object?

**Comment by tnoczyns-volue** [comment, marked as answer]
Hi,

The functionality you are asking for will be part of next Mesh release (planned as 2.6.1) and Mesh Python SDK release.
I'm currently working on this. I'll also add an example of how to traverse a model using the relation attributes.

Because in the Mesh version you are using the relations are supported in very limited way you can't read what objects are connected to them and only basic information (name, id and path) is returned as you correctly observed.

To give you a better understanding there are 2 groups of relations between objects in Mesh: ownership and link.

### Ownership relation

Each object is owned by some other object (owner). It is called ownership relation.

Ownership relations can be:
* one-to-one
* one-to-many (collection): where value of *many* is in range defined by *minimum* and *maximum cardinality* on relation `attribute definition` level.

By convention ownership relation attribute names start with 'has_...', e.g.: *has_PowerPlant*. Removing an object will remove recursively all of its owned objects as well, if there were any.

### Link relation

Link relation is a relation where one object may point to another object, but does not own it. There is also a versioned link relation, which is a link relation where the target object can change over time. It consists of a list of pairs:
* Target object identifier.
* Timestamp which indicates start of the period where the target object is active (linked to), the target object is active until the next target object in the list, if any, becomes active.

There are four types of link relation attributes:
* one-to-one
* one-to-many (collection)
* versioned one-to-one
* versioned one-to-many (collection)

Similar to `ownership relation` the value of *many* is in range defined by *minimum* and *maximum cardinality* on relation `attribute definition` level.

By convention link relation attribute names start with 'to_...', e.g.: 'to_EnergyMarket'. Removing an object that had link relations to other objects will NOT remove the linked objects.

**Comment by gajanthini** [comment]
Hi, 

I see. Nice to now that it will be implemented soon. And thank you for the thorough explanation of the two types of relations:)
